docs/asciidoc/modules/ROOT/pages/cypher-execution/cypher-based-procedures-functions.adoc

```diff
@@ -186,4 +130,88 @@ changes to each cluster member
+----
+
+Given statement will be registered as a procedure, the results will be turned into a stream of records.
+
+.Parameters
+[%autowidth,opts=header,cols="m,m,a"]
+|===
+| name
+| default
+| description
+
+| name | none | dot-separated name, will be prefixed with `custom`
+| statement | none | cypher statement to run, can use $parameters
+| mode | read | execution mode of the procedure: read, write, or schema
+| outputs | [["row","MAP"]] | List of pairs of name-type to be used as output columns, need to be in-order with the cypher statement, the default is a special case, that will collect all columns of the statement result into a map
+| inputs | [["params","MAP","{}"]] | Pairs or triples of name-type-default, to be used as input parameters. The default just takes an optional map, otherwise they will become proper paramters in order
+| description | "" | A general description about the business rules implemented into the procedure
+|===
+
+The type names can be the same as `apoc.custom.declare*` analogues.
+The default values are parsed as JSON.
+
+.Find neighbours of a node by name
+[source,cypher]
+----
+CALL apoc.custom.asProcedure('neighbours',
+  'MATCH (n:Person {name:$name})-->(nb) RETURN nb as neighbour','read',
+  [['neighbour','NODE']],[['name','STRING']], 'get neighbours of a person');
+
+CALL custom.neighbours('Keanu Reeves') YIELD neighbour;
+----
+
+
+
+== [DEPRECATED] Custom Functions with `apoc.custom.asFunction`
+
+The same is possible as a function:
+
+[source,cypher]
+----
+CALL apoc.custom.asFunction('answer','RETURN 42')
+----
+
+Given statement will be registered as a statement, the results into a single value.
+If the given output type is a list, results will be collected into a list, otherwise the first row will be used.
+The statement needs to return a single column, otherwise an error is thrown.
+
+.Parameters
+[%autowidth,opts=header, cols="m,m,a"]
+|===
+| name
+| default
+| description
+
+| name | none | dot-separated name, will be prefixed with `custom`
+| statement | none | cypher statement to run, can use $parameters
+| outputs | "LIST OF MAP" | Output type for single output, if the type is a list, then all rows will be collected, otherwise just the first row. Only single column results are allowed.
+If your single row result is a list you can force a single row by setting the last parameter to `true`
+| inputs | [["params","MAP","{}"]] | Pairs or triples of name-type-default, to be used as input parameters. The default just takes an optional map, otherwise they will become proper parameters in order
+| singleRow | false | If set to true, the statement is treated as single row even with the list result type, then your statement has to return a list.
+| description | "" | A general description about the business rules implemented into the function
+|===
+
+The type names are what you would expect and see in outputs of `dbms.procedures` or `apoc.help` just without the `?`.
+The default values are parsed as JSON.
```

Comment:
I would rather remove this part altogether and only show the non-deprecated way in this section